src/GafferScene/SetAlgo.cpp

```diff
@@ -449,6 +449,22 @@ IECore::MurmurHash setExpressionHash( const std::string &setExpression, const Sc
 	return h;
 }
 
+bool affectsSetExpression( const Plug *scenePlugChild )
+{
+	const ScenePlug *parent = scenePlugChild->parent<ScenePlug>();
+	if( !parent || parent->direction() != Gaffer::Plug::In )
```

Comment:
I don't think we need the direction check here, since set expressions won't always be used only internally to a node.  For instance, a UI element might be monitoring an output plug to show the results of a set expression.